plutus-use-cases/src/Plutus/Contracts/GameStateMachine.hs

```diff
@@ -163,8 +179,12 @@ data GameInput =
     deriving anyclass (ToJSON, FromJSON)
 
 {-# INLINABLE transition #-}
-transition :: State GameState -> GameInput -> Maybe (TxConstraints Void Void, State GameState)
-transition State{stateData=oldData, stateValue=oldValue} input = case (oldData, input) of
+transition
+    :: GameParam
```

Comment:
Shouldn't the `GameParam` be used here, or anywhere else? (Maybe I missed it)

Reply:
It's actually not used specifically anywhere. I only use it to parameterize the contract (i.e. each game will get a specific script address depending on the input value of `GameParam`). I'll look at if it makes sense to add validation for the `GameParam` input.

Reply:
Ah that makes sense. Maybe add a bit of documentation that the point is to get different script addresses.